app/decorators/gobierto_plans/indicator_data_decorator.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module GobiertoPlans
+  class IndicatorDataDecorator < BaseDecorator
+
+    attr_reader :plan, :collection
+
+    def initialize(plan)
+      @plan = plan
+      @collection = plan.indicators
+      @columns = 0
+    end
+
+    def csv
+      lines = []
+      CSV.generate do |csv|
+        collection.each do |item|
+          lines.concat(csv_row(item))
+        end
+        # need iterate again because header length is dynamic
+        csv << csv_header
+        lines.each do |line|
+          csv << line.concat(Array.new(csv_header.size - line.size, nil))
+        end
+      end.force_encoding('utf-8')
+    end
+
+    private
+
+    def csv_header
+      header = %w(Node.Title Node.external_id custom_field)
+      1.upto(@columns) do |index|
+        header << "col_#{index}"
+      end
+      header
+    end
+
+    def values_keeping_order(object,record)
+      attributes_in_order = object.custom_field.options.dig("configuration","plugin_configuration","columns").map { |r| r["id"] }
+      attributes_in_order.map do |key|
+        record[key]
+      end
+    end
+
+    def csv_row(object)
+      current_columns = object.payload[object.custom_field.uid.to_s].first&.size || 0
```

Comment:
Why not use `object.value`? This method is defined for custom field records [there](https://github.com/PopulateTools/gobierto/blob/4e6941f13b2bf37af73c2a7962e207d450eb1a4b/app/models/gobierto_common/custom_field_value/plugin.rb#L16-L20)